onnxruntime/core/providers/cpu/ml/tree_ensemble_aggregator.h

```diff
@@ -23,9 +23,18 @@ struct TreeNodeElementId {
   }
   struct hash_fn {
     std::size_t operator()(const TreeNodeElementId& key) const {
-      std::size_t h1 = std::hash<int64_t>()(key.tree_id);
-      std::size_t h2 = std::hash<int64_t>()(key.node_id);
-      return h1 ^ h2;
+      // Employs Elegant Pairing function to produce a unique non-negative integer from two non-negative integers.
+      // Reference: http://szudzik.com/ElegantPairing.pdf
+
+      u_int64_t combined_id;
+      u_int64_t x = static_cast<u_int64_t>(key.tree_id);
+      u_int64_t y = static_cast<u_int64_t>(key.node_id);
```

Comment:
It seems negative Id's aren't common. However, if you prefer to have a more predictable behaviour for negatives ID's we can use the old hash function for them or use the absolute values before casting to an unsigned integer.